### PR TITLE
login and signup features

### PR DESCRIPTION
**Pull request by alanDevgurus**
This PR includes changes needed to integrate the Commercetools provider.

It allow to use SignUp, Login (saving cookies of customerId) and access to Profile view. The changes included here correspond to the framework approach, later new changes will be applied with the use of the "me" of commercetools.

Some code has been temporarily commented on the components to allow the project build and execution while we work on the missing required hooks, this commit should be reverted once we implement all the underlying functionalities.

### CODE REVIEW

File: framework/commercetools/api/endpoints/customer/get-logged-in-customer.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    if (id) {
+      const { data } = await config.fetch<any>(getCustomerQuery, {
+        variables: { id },
+      })
+
+      const { customer } = data
+
+      if (!customer) {
+        return res.status(400).json({
```
Comment (leansoler) [inline framework/commercetools/api/endpoints/customer/get-logged-in-customer.ts]: shouldn't be 404?

File: framework/commercetools/api/operations/login.ts [hunk 2]
```diff
@@ -3,41 +3,64 @@ import type {
...
   }: {
     query?: string
     variables: T['variables']
     res: ServerResponse
     config?: Partial<CommercetoolsConfig>
   }): Promise<T['data']> {
     const config = commerce.getConfig(cfg)
-    return {
-      result: '',
+    const expireTime = new Date(Date.now() + 30 * 30) // 1 month
```
Comment (leansoler) [inline framework/commercetools/api/operations/login.ts]: This looks like duplicated code

File: framework/commercetools/api/endpoints/customer/get-logged-in-customer.ts (hunk 1 above)
Comment (alanDevgurus) [reply framework/commercetools/api/endpoints/customer/get-logged-in-customer.ts]: Yes. Done.

File: framework/commercetools/api/operations/login.ts (hunk 2 above)
Comment (alanDevgurus) [reply framework/commercetools/api/operations/login.ts]: Resolved

**Review by leansoler — APPROVED** [review]

**Pull request merged**